File: app/src/main/java/bassamalim/hidaya/fragments/QuranCollectionFragment.java
package bassamalim.hidaya.fragments;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.preference.PreferenceManager;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

import bassamalim.hidaya.R;
import bassamalim.hidaya.activities.QuranViewer;
import bassamalim.hidaya.databinding.FragmentCollectionQuranBinding;
import bassamalim.hidaya.dialogs.TutorialDialog;

public class QuranCollectionFragment extends Fragment {

    private FragmentCollectionQuranBinding binding;

    @Nullable @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        binding = FragmentCollectionQuranBinding.inflate(getLayoutInflater());

        checkFirstTime();

        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        QAdapter qAdapter = new QAdapter(this);
        ViewPager2 viewPager = view.findViewById(R.id.quran_pager);
        viewPager.setAdapter(qAdapter);

        TabLayout tabLayout = view.findViewById(R.id.tab_layout);

        String[] tabs = new String[] {getString(R.string.all), getString(R.string.favorite)};
        new TabLayoutMediator(tabLayout, viewPager,
                (tab, position) -> tab.setText(tabs[position])
        ).attach();
    }

    @Override
    public void onResume() {
        super.onResume();
        setupContinue();
    }

    private void setupContinue() {
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(requireContext());
        int page = pref.getInt("bookmarked_page", -1);
        String text = pref.getString("bookmarked_text", "");

        if (page == -1)
            text = getString(R.string.no_bookmarked_page);
        else {
            text = getString(R.string.bookmarked_page) + text;

            binding.continueReading.setOnClickListener(v -> {
                Intent intent = new Intent(getContext(), QuranViewer.class);
                intent.setAction("by_page");
                intent.putExtra("page", page);
                requireContext().startActivity(intent);
            });
        }

        binding.continueReading.setText(text);
    }

    private void checkFirstTime() {
        String key = "is_first_time_in_quran_fragment";
        if (PreferenceManager.getDefaultSharedPreferences(requireContext())
                .getBoolean(key, true))
            new TutorialDialog(getContext(),
                    getString(R.string.quran_fragment_tips), key).show(
                    requireActivity().getSupportFragmentManager(), TutorialDialog.TAG);
    }
}

class QAdapter extends FragmentStateAdapter {
    public QAdapter(Fragment fragment) {
        super(fragment);
    }

    @NonNull @Override
    public Fragment createFragment(int position) {
        Fragment fragment;

        if (position == 0)
            fragment = new AllQuranFragment();
        else
            fragment = new FavoriteQuranFragment();

        return fragment;
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/bassamalim/hidaya/fragments/QuranCollectionFragment.kt
package bassamalim.hidaya.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.preference.PreferenceManager
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.viewpager2.widget.ViewPager2
import bassamalim.hidaya.R
import bassamalim.hidaya.activities.QuranViewer
import bassamalim.hidaya.databinding.FragmentCollectionQuranBinding
import bassamalim.hidaya.dialogs.TutorialDialog
import bassamalim.hidaya.utils.DBUtils
import bassamalim.hidaya.utils.LangUtils
import bassamalim.hidaya.utils.PrefUtils
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayoutMediator

class QuranCollectionFragment : Fragment() {

    private var binding: FragmentCollectionQuranBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentCollectionQuranBinding.inflate(layoutInflater)

        checkFirstTime()

        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val qAdapter = QAdapter(this)
        val viewPager: ViewPager2 = view.findViewById(R.id.quran_pager)
        viewPager.adapter = qAdapter

        val tabLayout: TabLayout = view.findViewById(R.id.tab_layout)
        val tabs = arrayOf(getString(R.string.all), getString(R.string.favorite))
        TabLayoutMediator(tabLayout, viewPager) {
                tab: TabLayout.Tab, position: Int -> tab.text = tabs[position]
        }.attach()
    }

    override fun onResume() {
        super.onResume()
        setupContinue()
    }

    private fun setupContinue() {
        val pref = PreferenceManager.getDefaultSharedPreferences(requireContext())
        val page = pref.getInt("bookmarked_page", -1)
        val sura = pref.getInt("bookmarked_sura", -1)

        if (page == -1)
            binding!!.continueReading.text = getString(R.string.no_bookmarked_page)
        else {
            val db = DBUtils.getDB(requireContext())
            val suraName =
                if (PrefUtils.getLanguage(requireContext(), pref) == "en")
                    db.suarDao().getNameEn(sura)
                else db.suarDao().getName(sura)

            val text = "${getString(R.string.bookmarked_page)} ${getString(R.string.page)} " +
                    "${LangUtils.translateNums(requireContext(), page.toString())}," +
                    " ${getString(R.string.sura)} $suraName"

            binding!!.continueReading.setOnClickListener {
                val intent = Intent(context, QuranViewer::class.java)
                intent.action = "by_page"
                intent.putExtra("page", page)
                requireContext().startActivity(intent)
            }

            binding!!.continueReading.text = text
        }
    }

    private fun checkFirstTime() {
        val prefKey = "is_first_time_in_quran_fragment"
        if (PreferenceManager
                .getDefaultSharedPreferences(requireContext()).getBoolean(prefKey, true))
            TutorialDialog.newInstance(
                getString(R.string.quran_fragment_tips), prefKey
            ).show(requireActivity().supportFragmentManager, TutorialDialog.TAG)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

}

internal class QAdapter(fragment: Fragment?) : FragmentStateAdapter(fragment!!) {

    override fun createFragment(position: Int): Fragment {
        return if (position == 0) AllQuranFragment() else FavoriteQuranFragment()
    }

    override fun getItemCount(): Int {
        return 2
    }

}